daemon/images/images.go

```diff
@@ -202,20 +197,25 @@ func (i *ImageService) Images(imageFilters filters.Args, all bool, withExtraAttr
 					summary.Containers++
 				}
 			}
+			summaryMap[img] = summary
```

Comment:
Actually wondering now; if summary is unique per image, I _think_ we can skip this whole `withExtraAttrs` block, and if `summaryMap[img] ` already exists, we can instead pick the existing summary from `imagesMap[img]`, and add it to `summaries` ? Something like

```go
if withExtraAttrs {
    if s, ok := summaryMap[img]; ok {
        // Already calculated size for this image
        summaries = append(summaries, summary)
        continue
    }
}
```

(double check if I'm right though)

Reply:
Oh! I don't think that's correct; as this will be incrementing usage count 🤔

Reply:
Dang. Looking at the old code (which did `layerRefs[chid]++` in this part); I think for the new code, it may be correct (we can possibly reuse the cache).

Man, this code is confusing 😂